### DISCUSSION TITLE
Attacks with payloads functionality when you have more than 1 request

### DISCUSSION THREAD

**Discussion by pakkers** [projectdiscovery/nuclei-templates#2797] [Q&A]
Good afternoon, 

I've a question regarding how it works (or run) attacks with payloads when you have more than 1 request. This is the scenario:

  - raw:
#REQ1
      - |
        GET /miapp/home HTTP/1.1
        <some_headers>
#REQ2
      - |
        POST /miapp/j_security_check HTTP/1.1
        <some_headers>
        Cookie: §cookiesSESSION§

        username=admin&password={{payload}}

   payloads:
      pass: passDictionary.txt
    attack: sniper

With this scenario, I was waiting to see in my proxy 
1. REQ1
2. REQ2 using payload1
...
3. REQ1
4. REQ2 using payload1_n

But I get this

1. REQ1
2. REQ1 ...
3. REQ1 N
...
4. REQ2
5. REQ2
6. REQ2 using payload1_n

Seems that the attack is applied request by request independently and not like a one set of request.

What I'm doing bad? Or what is the way to run one attack over a set of request??

**Comment by olearycrew** [comment]
I'm not sure I understand what you're looking for here - for the request to overlap in parallel or be run in sequence serially. You may want to look at Nuclei workflows as a solution here, but I'm not 100% sure on that.
